src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateCommandRunner.cs

```diff
@@ -0,0 +1,261 @@
+                CacheContext = cacheContext,
+                LockFileVersion = LockFileFormat.Version,
+                Log = logger,
+                MachineWideSettings = new XPlatMachineWideSetting(),
+                GlobalPackagesFolder = globalPackagesFolder,
+                PreLoadedRequestProviders = providers
+                // Sources : No need to pass it, because SourceRepositories contains the already built SourceRepository objects
+            };
+
+            // Generate Restore Requests. There will always be 1 request here since we are restoring for 1 project.
+            var restoreRequests = await RestoreRunner.GetRequests(restoreContext);
+
+            // Run restore without commit. This will always return 1 Result pair since we are restoring for 1 request.
+            var restoreResult = await RestoreRunner.RunWithoutCommitAsync(restoreRequests, restoreContext, cancellationToken);
+
+            return restoreResult.Single();
+        }
+
+        private static async Task<(PackageDependency?, List<NuGetFramework>?)> GetPackageToUpdateAsync(
+            IReadOnlyList<string>? packages,
+            PackageSpec project,
+            CachingSourceProvider sourceProvider,
+            ISettings settings,
+            SourceCacheContext sourceCacheContext,
+            ILoggerWithColor logger,
+            CancellationToken cancellationToken)
+        {
+            if (packages is null || packages.Count == 0)
+            {
+                logger.LogMinimal("Unsupported: Must provide a package to update", ConsoleColor.Red);
+                return (null, null);
+            }
+
+            if (packages.Count > 1)
+            {
+                logger.LogMinimal("Unsupported: Updating more than one package is not yet supported", ConsoleColor.Red);
+                return (null, null);
+            }
+
+            var sourceMapping = PackageSourceMapping.GetPackageSourceMapping(settings);
+            if (sourceMapping.IsEnabled)
+            {
+                logger.LogMinimal("Projects using Package Source Mapping is not yet supported.", ConsoleColor.Red);
+                return (null, null);
+            }
+
+            var packageId = packages.Single();
+
+            VersionRange? existingVersion = null;
+            List<NuGetFramework>? frameworks = null;
+            foreach (var tfm in project.TargetFrameworks)
+            {
+                foreach (var dependency in tfm.Dependencies)
+                {
+                    if (string.Equals(packageId, dependency.Name, StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (frameworks is null)
+                        {
+                            frameworks = new List<NuGetFramework>();
+                        }
+                        frameworks.Add(tfm.FrameworkName);
+
+                        VersionRange tfmVersionRange;
+                        if (project.RestoreMetadata.CentralPackageFloatingVersionsEnabled)
```

Comment:
This code will need to work with `VersionOverride` as well.

Reply:
I forgot to create a tracking issue for this. It's planned for a future improvement, rather than the first version:
* https://github.com/NuGet/Home/issues/14318